pkg/auth/providers/azure/clients/ms_graph_client.go

```diff
@@ -151,14 +153,19 @@ func msGraphTokenHasPermissions(token string) bool {
 	if !ok {
 		return false
 	}
-	roleNames := make([]string, len(roles))
+	permissionPresent := make(map[string]bool)
+	for _, p := range msGraphRequiredPermissions {
+		permissionPresent[p] = false
```

Comment:
Nit:

You could just check if the permission is not present in the map at the end rather than re-initing to false here.

Marking this as a nit because the efficiency loss is so minor.